File: ADF1/Assignment 7/News.java
import java.util.ArrayList;

public class News implements INews {
    int id;
    String title;
    String publishDate;
    String author;
    String content;
    float averageRate = 0;
    int[] rateList = new int[3];

    public News(String title, String publishDate, String author, String content) {
        this.title = title;
        this.publishDate = publishDate;
        this.author = author;
        this.content = content;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPublishDate() {
        return publishDate;
    }

    public void setPublishDate(String publishDate) {
        this.publishDate = publishDate;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public float getAverageRate() {
        return averageRate;
    }

    public int[] getRateList() {
        return rateList;
    }

    @Override
    public void display() {
        System.out.println("Title: " + title);
        System.out.println("PublishDate: " + publishDate);
        System.out.println("Author: " + author);
        System.out.println("Content: " + content);
        System.out.println("AverageRate: " + averageRate);
    }

    public void calculate() {
        averageRate = 0;
        for (int i = 0; i < rateList.length; i++) {
            averageRate += rateList[i];
        }
        averageRate = averageRate / 3;
    }
}

File: CSharp/Assignment4/News.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignment4
{
    class News : INews
    {
        private int id;
        private string title;
        private string publishDate;
        private string author;
        private string content;
        private float averageRate;

        public int Id { get; set; }
        public string Title { get; set; }
        public string PublishDate { get; set; }
        public string Author { get; set; }
        public string Content { get; set; }
        public float AverageRate { get; }

        public void Display()
        {
            Console.WriteLine("Id: " + this.id);
            Console.WriteLine("Title: " + this.title);
            Console.WriteLine("PublishDate: " + this.publishDate);
            Console.WriteLine("Author: " + this.author);
            Console.WriteLine("Content: " + this.content);
            Console.WriteLine("AverageRate: " + this.averageRate);
            Console.WriteLine();
        }

        int [] RateList = new int [3];

        public void Calculate()
        {
            int total = 0;
            for (int i = 0; i < RateList.Length; i++)
            {
                total += RateList[i];
            }
            this.averageRate = total / 3;
        }

        public News()
        {
        }

        public News(string title, string publishDate, string author, string content, int[] rateList)
        {
            this.title = title;
            this.publishDate = publishDate;
            this.author = author;
            this.content = content;
            RateList = rateList;
        }
    }
}

